workflow/controller/workflowpod.go

```diff
@@ -332,12 +333,27 @@ func (woc *wfOperationCtx) createWorkflowPod(nodeName string, mainCtr apiv1.Cont
 			return nil, errors.Wrap(err, "", "Error in Unmarshalling after merge the patch")
 		}
 	}
+
+	// we must check to see if the pod exists rather than just optimistically creating the pod and see if we get
+	// an `AlreadyExists` error because we won't get that error if there is not enough resources
+	obj, exists, err := woc.controller.podInformer.GetStore().Get(cache.ExplicitKey(pod.Namespace + "/" + pod.Name))
```

Comment:
Can we do this check when the forbidden error happens? This will help to avoid the unnecessary API  calls

Reply:
this is done before that happens, it is the `pod.Create(..)` that return forbidden